### PR TITLE
Drop Python 3.9 support in favor of 3.12 and 3.13

### PR DESCRIPTION
**Pull request by thalassemia**
I was able to update `requirements.txt` to work on Python 3.10-3.13. Unfortunately, the latest possible version of `networkx` that installs on Python 3.9 is 3.2.1, and a change in `networkx` 3.4 breaks our `plot_topology` function. I could add logic to determine the version of `networkx` and do different things depending on whether it is greater or less than 3.4, but that sounds unnecessarily complicated. Instead, I think it makes more sense to just bump the required Python version to 3.10 and `networkx` version to 3.4. For context, Python 3.9 reaches end-of-life in 2025-10, and big packages like Numpy and Scipy have already dropped support for it.

Other changes include:
- Assorted pylint and mypy fixes
- Change default multiprocessing start method to ``forkserver`` on Linux to fix python/cpython#110770 (default will be ``forkserver`` in Python 3.14 anyways)

-----------------------------------------------------------------------

By creating this pull request, I agree to the Contributor License
Agreement, which is available in `CLA.md` at the top level of this
repository.

### CODE REVIEW

File: vivarium/core/process.py [hunk 1]
```diff
@@ -717,8 +717,18 @@ def __init__(
...
+        # Linux's default ``fork`` start method causes a lot of random
+        # issues, including python/cpython#110770 (prompted this change)
+        # and python/cpython#84559 (general discussion). This default
+        # will be changed to ``forkserver`` in Python 3.14. MacOS and
+        # Windows use the much safer but slightly slower ``spawn`` method
+        if sys.platform not in ("darwin", "win32"):
+            start_method = "forkserver"
+        else:
+            start_method = "spawn"
+        mp_ctx = multiprocessing.get_context(start_method)
```
Comment (eagmon) [inline vivarium/core/process.py:729]: great!

**Review by eagmon — APPROVED** [review]
Thanks @thalassemia!

File: doc/requirements.txt [hunk 2]
```diff
@@ -1,87 +1,105 @@
...
+matplotlib-inline==0.1.7
+mccabe==0.7.0
+mistune==3.0.2
+mypy==1.13.0
+mypy-extensions==1.0.0
+nbclient==0.10.1
+nbconvert==7.16.4
+nbformat==5.10.4
+nbsphinx==0.9.5
+networkx==3.4.2
```
[the remark this answers is not in the document]
Comment (thalassemia) [reply doc/requirements.txt:50]: These are the current latest versions of all requirements that work on 3.10-3.13. The problem with 3.9 is that you can't install networkx 3.4+, which requires a change in plot_topology to work (included in this PR).

**Comment by thalassemia** [comment]
> Thanks @thalassemia! So is there topology graphing as it currently stands?

plot_topology works in the current release as long as you have networkx<3.4. I think those older versions of networkx should work on Python 3.10-3.13 as well. I don't think it is a good idea to cap the networkx version to address the issue with networkx 3.4+.

**Comment by eagmon** [comment]
@thalassemia -- looks like there are still some 3.9-related tests hanging. Can you remove these from the GitHub actions?

**Comment by thalassemia** [comment]
I think those are required by some repository setting that I can't access. Instead of hanging they're just not being run. [These docs](https://docs.github.com/en/enterprise-server@3.11/actions/sharing-automations/required-workflows) might be helpful.

**Pull request merged**